src/Runtime/LambdaRuntime.php

```diff
@@ -131,16 +145,19 @@ private function waitNextInvocation(): array
             $this->closeHandler();
             throw new \Exception('Failed to fetch next Lambda invocation: ' . $message);
         }
-        if ($invocationId === '') {
-            throw new \Exception('Failed to determine the Lambda invocation ID');
-        }
         if ($body === '') {
             throw new \Exception('Empty Lambda runtime API response');
         }
 
+        $context = $contextBuilder->buildContext();
+
+        if ($context->getAwsRequestId() === '') {
```

Comment:
Is a Context with an empty aws request id valid? Maybe this check should be moved in the ContextBuilder? (same question for the other properties of the context, what is the minimal set of data that constitutes a valid context?)

Reply:
context can be completely empty but that piece of code expects a non-empty invocation id. It's just a coincidence I'm refetching it from the context instead of keeping around in an $invocationId variable. I could do revert back to using a $invocationId variable in that piece of code and build the context last just before return

Reply:
It might ideally be better indeed to forbid having an invalid Context around, but honestly this should never happen and an exception will be thrown if it ever happens. Let's not block this PR longer than necessary.

Thank you for updating your PR @amne.